src/plugins/node-workspace.ts

```diff
@@ -0,0 +1,412 @@
+        updated.version
+      );
+    }
+  }
+
+  private getChangelogDepsNotes(
+    pkg: Package,
+    origPkgJson: PackageJson
+  ): string {
+    let depUpdateNotes = '';
+    type DT =
+      | 'dependencies'
+      | 'devDependencies'
+      | 'peerDependencies'
+      | 'optionalDependencies';
+    const depTypes: DT[] = [
+      'dependencies',
+      'devDependencies',
+      'peerDependencies',
+      'optionalDependencies',
+    ];
+    const updates: Map<DT, string[]> = new Map();
+    for (const depType of depTypes) {
+      const depUpdates = [];
+      const pkgDepTypes = pkg[depType];
+      if (pkgDepTypes === undefined) {
+        continue;
+      }
+      for (const [depName, currentDepVer] of Object.entries(pkgDepTypes)) {
+        const origDepVer = origPkgJson[depType]?.[depName];
+        if (currentDepVer !== origDepVer) {
+          depUpdates.push(
+            `\n    * ${depName} bumped from ${origDepVer} to ${currentDepVer}`
+          );
+        }
+      }
+      if (depUpdates.length > 0) {
+        updates.set(depType, depUpdates);
+      }
+    }
+    for (const [dt, notes] of updates) {
+      depUpdateNotes += `\n  * ${dt}`;
+      for (const note of notes) {
+        depUpdateNotes += note;
+      }
+    }
+    return depUpdateNotes;
+  }
+
+  private updateChangelogEntry(
+    exChangelog: string,
+    changelogEntry: string,
+    pkg: Package
+  ): string {
+    changelogEntry = changelogEntry.replace(
+      new RegExp(`^###? \\[${pkg.version}\\].*### Dependencies`, 's'),
+      '### Dependencies'
+    );
+    const match = exChangelog.match(
+      new RegExp(
+        `(?<before>^.*?###? \\[${pkg.version}\\].*?\n)(?<after>###? [0-9[].*)`,
+        's'
+      )
+    );
+    if (!match) {
+      this.log(
+        `Appending update notes to end of changelog for ${pkg.name}`,
+        CheckpointType.Failure
+      );
+      changelogEntry = `${exChangelog}\n\n\n${changelogEntry}`;
+    } else {
+      const {before, after} = match.groups!;
+      changelogEntry = `${before.trim()}\n\n\n${changelogEntry}\n\n${after.trim()}`;
+    }
+    return changelogEntry;
+  }
+
+  private async newChangelogEntry(
+    changelogEntry: string,
+    changelogPath: string,
+    pkg: Package
+  ): Promise<string> {
+    let changelog;
+    try {
+      changelog = (await this.gh.getFileContents(changelogPath)).parsedContent;
+    } catch (e) {
+      if (e.status !== 404) {
+        this.log(
+          `Failed to retrieve ${changelogPath}: ${e}`,
+          CheckpointType.Failure
+        );
+        return '';
+      } else {
+        this.log(
+          `Creating a new changelog at ${changelogPath}`,
+          CheckpointType.Success
+        );
+      }
+    }
+    const changelogUpdater = new Changelog({
+      path: changelogPath,
+      changelogEntry,
+      version: pkg.version,
+      packageName: pkg.name,
+    });
+    return changelogUpdater.updateContent(changelog);
+  }
+
+  private async setChangelogEntry(
```

Comment:
all this changelog entry manipulation got a bit exasperating at the end but it seems to work pretty well now. There may be a better way of doing it but I was just trying to jam through to the finish line... really wanted to see changelog entries for lerna-based dependency updates.

Reply:
See comment about us moving towards a better markdown generator that actually understands the tree.